ast/builtins.go

```diff
@@ -1030,6 +1031,20 @@ var JWTDecodeVerify = &Builtin{
 	),
 }
 
+// JWTEncodeSign encodes and optionally sign  a JSON Web Token.
+// Inputs are protected headers, payload, secret
+var JWTEncodeSign = &Builtin{
+	Name: "io.jwt.encode_sign",
+	Decl: types.NewFunction(
+		types.Args(
+			types.S,
```

Comment:
I'm curious why the parameters are passed as JSON serialized strings and not just JSON values. Let's address this (because it's the interface, a bunch of the rest of the code) and then move onto the rest of the code.

Reply:
Because I copied them directly from code for the example. Works with regular JSON strings as well, like so:

> io.jwt.encode_sign(`{"typ":"JWT",
|  "alg":"HS256"}`, `{"iss":"joe",
|  "exp":1300819380,
|  "http://example.com/is_root":true}`, `{
| "kty":"oct",
| "k":"AyM1SysPpbyDfgZld3umj1qzKObwVMkoqQ-EstJQLr_T-1qS0gZH75aKtMN3Yj0iPS4hcgUuTwjAzZr1Z9CAow"
| }`,x)
|
+-------------------------------------------------------------------------------------+
|                                          x                                          |
+-------------------------------------------------------------------------------------+
| "eyJ0eXAiOiJKV1QiLAogImFsZyI6IkhTMjU2In0.eyJpc3MiOiJqb2UiLAogImV4cCI6MTMwMDgxOTM... |
+-------------------------------------------------------------------------------------+

Reply:
Github got confused and highlighted instead of preserving single quotes. There are single quotes around each JSON object.

Reply:
Whoops, I replied above accidentally. Re-posting here in the correct location.

@repenno I'm expecting the headers, payload, and JWK to be passed as JSON objects instead of requiring callers (i) JSON marshal values and pass or (ii) hardcode JSON serialized strings into their policy.